neverbleed.c

```diff
@@ -390,15 +396,61 @@ static RSA *daemon_get_rsa(size_t key_index)
     return rsa;
 }
 
-static size_t daemon_set_rsa(RSA *rsa)
+/*
+ *  Returns one plus the index of the least significant 1-bit of b within tot
+ *  or if not found, returns zero
+ */
+static uint32_t bita_ffirst(const uint8_t *b, const size_t tot, size_t bits)
```

Comment:
The function currently returns a `uint32_t` value that represents the true offset + 1, which seems a bit confusing to me.

Could we change the returned type to `size_t` (because the purpose of the value is to represent the offset of an array), and return the offset (or `SIZE_MAX` if not found) instead of doing +1?

Reply:
heh yeah it's repeating the behavior of __builtin_ffsll() which does not seem very popular.
Yeah good idea. We can adjust the index here as well. thanks!